frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -47,21 +57,63 @@ export const experimentLogic = kea<experimentLogicType>({
                 setExperimentFunnel: (_, { funnel }) => funnel,
             },
         ],
+        newExperimentCurrentPage: [
+            0,
+            {
+                nextPage: (page) => page + 1,
+                prevPage: (page) => page - 1,
+                setPage: (_, { page }) => page,
+            },
+        ],
     },
     listeners: ({ values, actions }) => ({
         createExperiment: async ({ draft }) => {
-            await api.create(`api/projects/${values.currentTeamId}/experiments`, {
-                ...values.newExperimentData,
-                ...(!draft && { start_date: dayjs() }),
-            })
-            experimentsLogic.actions.loadExperiments()
+            try {
+                if (values.newExperimentData?.id) {
+                    await api.update(`api/projects/${values.currentTeamId}/experiments/${values.experimentId}`, {
+                        start_date: dayjs(),
+                    })
+                } else {
+                    await api.create(`api/projects/${values.currentTeamId}/experiments`, {
+                        ...values.newExperimentData,
+                        ...(!draft && { start_date: dayjs() }),
+                    })
+                }
+            } catch (error) {
+                errorToast(
+                    'Error creating your experiment',
+                    'Attempting to create this experiment returned an error:',
+                    error.status !== 0
+                        ? error.detail
+                        : "Check your internet connection and make sure you don't have an extension blocking our requests.",
+                    error.code
+                )
+                return
+            }
+            toast.success(
+                <div data-attr="success-toast">
+                    <h1>Experimentation created successfully!</h1>
+                    <p>Click here to go back to the experiments list.</p>
+                </div>,
+                {
+                    onClick: () => {
+                        experimentsLogic.actions.loadExperiments()
+                        router.actions.push(urls.experiments())
+                    },
+                    closeOnClick: true,
+                }
+            )
         },
-        createNewExperimentFunnel: async () => {
+        createNewExperimentFunnel: async ({ filters }) => {
```

Comment:
We need to do this so when we open a draft experiment, the funnel shows up correctly?

Reply:
yes because we never save the funnel itself to the experiment, so we have to recreate it based on the filters from the experiment